src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -327,7 +331,10 @@ private ClusterDescription ProcessReplicaSetChange(ClusterDescription clusterDes
                     if (_maxElectionInfo != null)
                     {
                         isCurrentPrimaryStale = _maxElectionInfo.IsStale(args.NewServerDescription.ReplicaSetConfig.Version.Value, args.NewServerDescription.ElectionId);
-                        var isReportedPrimaryStale = !isCurrentPrimaryStale;
+                        var isReportedPrimaryStale = _maxElectionInfo.IsFresh(
```

Comment:
This change helps prevent us from unnecessarily updating the ElectionInfo when it's not needed. Previously, storing a new ElectionInfo with the same information was harmless, but now that we're logging when we update the ElectionInfo, we needed to change this to avoid unnecessarily filling the logs with spam.

Reply:
A bit of additional clarification:
More specifically, we care about two things here:
1. The driver wants to store the youngest tuple of (setVersion, electionId). 
2. The driver wants to invalidate a server if it reports a tuple that's older than what we have stored. 

If our stored tuple is not the youngest, then `isCurrentPrimaryStale` evaluates to true.
If the tuple we just received from the server (i.e. the reported tuple) is older than our stored tuple, then `isReportedPrimaryStale` evaluates to true.

Previously, in the implementation of `IsStale` we used a `<=` when comparing electionIds, resulting in equivalent tuples being harmlessly marked as stale (since the only consequence of this was storing the "new" tuple). The older implementation also allowed us to simply negate `isCurrentPrimaryStale` to determine if the reported primary is stale (concern #2 from above).

Reply:
It's a little confusing that IsStale and IsFresh are not exact opposites of each other, but I suppose that's inherent in the way the tuples are compared.

They can both return false under certain conditions right? But they can never both return true?